src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -2224,6 +2246,41 @@ export class StudyViewPageStore
         }
     }
 
+    parseRawFilterJson(filterJson: string): any {
+        let rawJson;
+        try {
+            rawJson = JSON.parse(decodeURIComponent(filterJson));
+        } catch (e) {
+            console.error('FilterJson invalid Json: error: ', e);
+        }
+        return rawJson;
+    }
+
+    fetchSamplesWithSampleListIds(sampleListIds: string[]) {
+        return defaultClient.fetchSamplesUsingPOST({
+            sampleFilter: {
+                sampleListIds: sampleListIds,
+            } as SampleFilter,
+            projection: 'SUMMARY',
+        });
+    }
+
+    convertPatientIdentifiersToSampleIdentifiers(
+        patientIdentifiers: Array<PatientIdentifier>,
+        samples: Sample[]
+    ): SampleIdentifier[] {
+        return samples
+            .filter(s =>
+                patientIdentifiers.some(
```

Comment:
as ino pointed out today, there could be a crapload of patients potentially, so what i would do is first, before loop, make a dictionary with the compound key p.patientId+p.studyId.   then you can use lookup (string in object) which is a zillion times faster than array scanning.